libs/ml/tests/unit/ops/avg_pool_1d.cpp

```diff
@@ -52,11 +52,13 @@ TYPED_TEST(AvgPool1DTest, forward_test_3_2_2)
   {
     for (SizeType i{0}; i < 10; ++i)
     {
-      data(0, i, i_b) = static_cast<DataType>(data_input[i]) + static_cast<DataType>(i_b * 10);
+      data(0, i, i_b) = fetch::math::AsType<DataType>(data_input[i]) +
+                        fetch::math::AsType<DataType>(static_cast<double>(i_b * 10));
```

Comment:
all of these tests that fill a tensor indirectly via a std::vector<double> seem pretty wrong to me. I think they can all be re-written only using Tensor<DataType> (which avoids this bizarre double casting)